syntax = "proto3";

package syft.core.node.common.action;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";

message SMPCActionMessage {
  string name_action = 1;
  syft.core.common.UID self_id = 2;
  repeated syft.core.common.UID args_id = 3;
  map<string, syft.core.common.UID> kwargs_id = 4;
  syft.core.common.UID id_at_location = 6;
  syft.core.io.Address address = 7;
  syft.core.common.UID msg_id = 8;
}
